source/macros.tex

```diff
@@ -275,12 +275,17 @@
 % ending with END TEXT. A counter with name ENVIRON indicates the
 % number of this kind of note / example that has occurred in this
 % subclause.
-\newcommand{\newnoteenvironment}[3]{
-\newsubclausecounter{#1}
-\newenvironment{#1}
-{\def\noteend{#3}\par\small\stepcounter{#1}\noteintro{#2}}
-{\noteoutro{\noteend}\par}
+\ExplSyntaxOn
+\NewDocumentCommand \newnoteenvironment { m m m } {
+  \newsubclausecounter{#1}
+  \NewDocumentEnvironment {#1*} { }
+    {\def\noteend{#3}\par\small\stepcounter{#1}\noteintro{#2}}
+    {\noteoutro{\noteend}}
+  \NewDocumentEnvironment {#1} { }
+    {\begin{note*}}
+    {\end{note*}\par}
```

Comment:
Should these be something like `#1*`?

Reply:
It absolutely should! I've fixed this. Thanks!